src/widgets/tabs.rs

```diff
@@ -121,17 +136,33 @@ where
                 (t, self.style, lt)
             }
         }) {
-            x += 1;
             if x >= tabs_area.right() {
                 break;
+            }
+
+            let mut space_remaining: isize = (tabs_area.right() as isize) - (x as isize);
```

Comment:
The code usually avoids dealing with signed integer to limit conversions. You can use saturing add and sub operations to avoid overflows.

Reply:
I'm not sure I understand what you mean by this.

Reply:
No I meant that usually I try to keep using u16 or usize because you can't represent impossible states (e.g negative width) and as a tendency to panic more often which makes the debugging experience easier and makes you more aware of the edge cases.

Reply:
Ah, alright I'll change that up. Thanks